bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  // Add funds to the boost pool
+  console.log(`Adding boost funds of ${amount} ${asset} at ${boostTier}bps`);
+  await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].addBoostFunds(
+      shortChainFromAsset(asset).toUpperCase(),
+      amountToFineAmount(amount.toString(), assetDecimals(asset)),
+      boostTier,
+    ),
+    lp,
+  );
+
+  return observeBoostFundsAdded;
+}
+
+/// Adds boost funds to the boost pool and does a swap with boosting enabled, then stops boosting and checks the fees collected are correct.
+async function testBoostingForAsset(asset: Asset, boostFee: number, lpUri: string, amount: number) {
+  await using chainflip = await getChainflipApi();
+  console.log(`Testing boosting for ${asset} at ${boostFee}bps`);
+
+  // Start with a clean slate by stopping boosting before the test
+  const preTestStopBoostingEvent = await stopBoosting(asset, boostFee, lpUri, false);
+  assert.strictEqual(
+    preTestStopBoostingEvent?.data?.pendingBoosts?.length ?? 0,
+    0,
+    'Stopped boosting but, the test cannot start with pending boosts.',
+  );
+
+  // Add boost funds
+  console.log(`Providing liquidity for boosting`);
+  await addBoostFunds(asset, boostFee, amount, lpUri);
+
+  // TODO: Check that we are the only booster in the pool
+
+  // Do a swap
+  const swapAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
+  const destAddress = await newAddress(swapAsset, 'LP_1');
+  console.log(`Swap destination address: ${destAddress}`);
+  const swapRequest = await requestNewSwap(
+    asset,
+    swapAsset,
+    destAddress,
+    undefined,
+    undefined,
+    0,
+    false,
+    boostFee,
+  );
+
+  const observeDepositFinalised = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:DepositFinalised',
+    chainflip,
+    (event) => event.data.channelId === swapRequest.channelId.toString(),
+  );
+  const observeSwapBoosted = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:DepositBoosted',
+    chainflip,
+    (event) => event.data.channelId === swapRequest.channelId.toString(),
+  );
+
+  await send(asset, swapRequest.depositAddress, amount.toString());
+  console.log(`Sent ${amount} ${asset} to ${swapRequest.depositAddress}`);
+
+  // Check that the swap was boosted
+  const depositEvent = await Promise.race([observeSwapBoosted, observeDepositFinalised]);
+  if (depositEvent.name.method === 'DepositFinalised') {
+    throw new Error('Deposit was finalised without seeing the DepositBoosted event');
+  } else if (depositEvent.name.method !== 'DepositBoosted') {
+    throw new Error(`Unexpected event ${depositEvent.name.method}`);
+  }
+
+  const depositFinalisedEvent = await observeDepositFinalised;
```

Comment:
I'm not sure how js handles this - have you tried to make it deliberately fail in this way? mainly thinking about the fact we have to copy the promise the here that was used above

Reply:
Yes I tested it and it works as intended.